bundled/CMakeLists.txt

```diff
@@ -28,11 +28,13 @@ IF(FEATURE_BOOST_BUNDLED_CONFIGURED)
 
   ADD_SUBDIRECTORY(${BOOST_FOLDER}/libs/serialization/src)
   ADD_SUBDIRECTORY(${BOOST_FOLDER}/libs/system/src)
+  ADD_SUBDIRECTORY(${BOOST_FOLDER}/libs/iostreams/src)
 
-  IF(DEAL_II_WITH_ZLIB)
-    ADD_SUBDIRECTORY(${BOOST_FOLDER}/libs/iostreams/src)
-  ELSE()
-    MESSAGE(STATUS "BOOST::Iostreams will not be available because zlib or its header files could not be found")
+  IF(DEAL_II_WITH_P4EST AND NOT DEAL_II_WITH_ZLIB)
+    MESSAGE(FATAL_ERROR
+            "The bundled Boost package requires "
+            "DEAL_II_WITH_ZLIB=ON to be enabled if "
+            "compiling with DEAL_II_WITH_P4EST=ON!")
```

Comment:
It would be much cleaner to express this dependency with
```
SET(FEATURE_P4EST_DEPENDS ZLIB)
```
in `configure_p4est.cmake`. Or, alternatively simply always enabling `DEAL_II_WITH_ZLIB` as we do for boost.

Reply:
(aka I am on a holy crusade to not introduce manual `FATAL_ERROR` in the configure process ^^)